xclim/sdba/utils.py

```diff
@@ -586,3 +586,124 @@ def best_pc_orientation(A, Binv, val=1000):
             # New orientation is better, keep and remember error.
             err = new_err
     return orient
+
+
+def get_clusters_1d(data: np.ndarray, u1: float, u2: float):
+    """Get clusters of a 1D array.
+
+    A cluster is defined as a sequence of values larger than u2 with at least one value larger than u1.
+
+    Inpsired from julia's Extremes package, function getcluster.
+
+    Parameters
+    ----------
+    data: 1D ndarray
+      Values to get clusters from.
```

Comment:
I'm worried using the term "cluster" will create confusion. We're using cluster in the ensembles function, and here it is used to mean something else entirely. 

I suspect this is the word used in the Julia code, but I suggest once Phil has reviewed the PR, we switch the name to something else. 

Elsewhere we use the term "run" or "sequence", no ? We could call those evs (extreme value sequences) ?

Related question: why does it matter that these are sequences ? In my mind, the Pareto describes the distribution of `x - thresh` for `x > thresh`, so it's not clear where `u1` comes from, and why the concept of sequence/clusters is needed.

Reply:
`clusters` word comes mainly from Coles (2001) and is the appropriate word in the statistical context (but not within the xclim nomenclature). 

The usefulness of declustering comes from the assumption that the convergence assumes independence between extracted exceedence. This assumption does probably not hold if you get multiple exceedence from a single weather event. So the clusters are defined by a lower threshold and we only extract the maximum value within each cluster.

Perhaps this assumption can be relaxed though and be linked to a keyword argument `decluster=True`.

Reply:
Ok, je comprend mieux ! On garde ça comme ça.